src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -50,11 +50,16 @@ public Response<List<PlaceholderNameAndValue>> Find(string path, string query, s
 
                     return new OkResponse<List<PlaceholderNameAndValue>>(placeHolderNameAndValues);
                 }
-                else if (IsCatchAll(path, counterForPath, pathTemplate))
+                else if (IsCatchAll(path, counterForPath, pathTemplate) || (pathTemplate[counterForTemplate] == '{') && NoMoreForwardSlash(pathTemplate, counterForTemplate) && NotPassedQueryString(pathTemplate, pathTemplate.Length))
                 {
                     var endOfPlaceholder = GetNextCounterPosition(pathTemplate, counterForTemplate, '}');
 
-                    var placeholderName = GetPlaceholderName(pathTemplate, 1);
+                    if (pathTemplate[counterForTemplate] == '/')
+                    {
+                        counterForTemplate++;
+                    }
```

Comment:
What do you fix here?
What case with what path?

Reply:
this is the Case when to handle the following 
/{placeholder} => /{placeholder} 
to handle the case of : /{emptystring} => /{emptystring}